kcl-ciris/src/main/scala/kinesis4cats/kcl/fs2/ciris/KCLCirisFS2.scala

```diff
@@ -118,29 +124,45 @@ object KCLCirisFS2 {
       metricsFactory: Option[MetricsFactory] = None,
       glueSchemaRegistryDeserializer: Option[GlueSchemaRegistryDeserializer] =
         None,
-      encoders: RecordProcessor.LogEncoders = RecordProcessor.LogEncoders.show
+      encoders: RecordProcessor.LogEncoders = RecordProcessor.LogEncoders.show,
+      managedClients: Boolean = true
   )(implicit
       F: Async[F],
       P: Parallel[F]
-  ): Resource[F, KCLConsumerFS2[F]] = kclConfig(
-    kinesisClient,
-    dynamoClient,
-    cloudwatchClient,
-    prefix,
-    shardPrioritization,
-    workerStateChangeListener,
-    coordinatorFactory,
-    customShardDetectorProvider,
-    tableCreatorCallback,
-    hierarchicalShardSyncer,
-    leaseManagementFactory,
-    leaseExecutorService,
-    aggregatorUtil,
-    taskExecutionListener,
-    metricsFactory,
-    glueSchemaRegistryDeserializer,
-    encoders
-  ).map(new KCLConsumerFS2[F](_))
+  ): Resource[F, KCLConsumerFS2[F]] = for {
+    kClient <-
+      if (managedClients)
+        Resource.fromAutoCloseable(
+          F.delay(kinesisClient)
+        )
+      else Resource.pure[F, KinesisAsyncClient](kinesisClient)
+    dClient <-
+      if (managedClients) Resource.fromAutoCloseable(F.delay(dynamoClient))
+      else Resource.pure[F, DynamoDbAsyncClient](dynamoClient)
+    cClient <-
+      if (managedClients)
+        Resource.fromAutoCloseable(F.delay(cloudWatchClient))
+      else Resource.pure[F, CloudWatchAsyncClient](cloudWatchClient)
```

Comment:
This and similar approaches were used to create default clients, but also allow users to manage their own resources if needed via setting `managedClients` to false